src/main/java/nl/topicus/bitbucket/api/PullRequestListener.java

```diff
@@ -207,7 +208,12 @@ private void sendPullRequestEvent(PullRequestEvent event, EventType eventType, b
             {
                 try
                 {
-                    pullRequestService.canMerge(pullRequest.getToRef().getRepository().getId(), pullRequest.getId());
+                    ScmPullRequestCommandFactory pullRequestCommandFactory = scmService.getPullRequestCommandFactory(pullRequest);
+                    // Require 4.10 to avoid deprecation warning.
+                    if (Version.check(applicationPropertiesService.getBuildVersion(), ">= 4.10.0"))
+                        pullRequestCommandFactory.tryMerge(pullRequest);
+                    else
+                        pullRequestCommandFactory.canMerge();
                 }
                 catch (IllegalPullRequestStateException e)
```

Comment:
With this change, `IllegalPullRequestStateException` can no longer happen, so this `catch` block is now redundant. However, rather than fully removing it, I'd probably change it to something like this:

```java
catch (ServiceException e) {
    LOGGER.warn("{}: Mergeability check failed; pull request refs may not be up-to-date", pullRequest, e);
}
```

With this exception, I _would_ log the full exception, because this `Command` should _never actually fail_. Even if the merge itself fails, that should be in the result of the `call()`, rather than an exception.